src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/Regex.Groups.Tests.cs

```diff
@@ -958,6 +958,16 @@ private static IEnumerable<(CultureInfo Culture, string Pattern, string Input, R
                 yield return (enUS, @"(?((\w{3}))\1\1|no)", "no", RegexOptions.None, new string[] { "no", "" });
             }
 
+            // Special cases involving starting position search optimizations
+            yield return (enUS, @"(\d*)(hello)(\d*)", "123hello456", RegexOptions.None, new string[] { "123hello456", "123", "hello", "456" });
+            yield return (enUS, @"((\d*))[AaBbCc](\d*)", "1b", RegexOptions.None, new string[] { "1b", "1", "1", "" });
```

Comment:
why the nested captures in particular? 

is it worth doing a few other variations, eg., `@"(\d*)(hello)(\d*)", "hello"` `@"(\w*)(hello)(\w*)", "hello"`.. no reason you regressed these, but basically throwing in 5-10 variations while you're here

Reply:
> why the nested captures in particular?

Because the implementation traverses down through nested captures and atomics.